### ISSUE TITLE
[2020] Dialogue Response Ranking Training with Large-Scale Human Feedback Data

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#115]
## 0. 論文

| タイトル | Dialogue Response Ranking Training with Large-Scale Human Feedback Data |
|:-------|:--|
| リンク  | https://arxiv.org/abs/2009.06978 |
| 著者    | Xiang Gao, Yizhe Zhang, Michel Galley, Chris Brockett, Bill Dolan (Microsoft Research) |
| 投稿日付 | EMNLP 2020 |

## 1. どんなもの？

最近の対話モデルはコンテキストをふまえた人間らしい返答をできるようになってきているが、より人同士の対話(相手を引きつけ、相互交流を促進するようなもの; engagingness)に近い返答を実現するために、2つの返答についてその魅力をランキング問題として解く GPT-2 ベースのモデル DialogRPT (Ranking Pretrained Transformers) を提案した。
ソーシャルメディアの投稿の返信数やツリーの長さ、upvote 数などの "feedback" データを用いている。
![スクリーンショット 2020-12-03 2 00 52](https://user-images.githubusercontent.com/1255485/100905460-70e68500-350b-11eb-869b-39414a4a67cf.png)
ベースラインモデルよりも人による評価値との高い相関を確認した。

## 2. 先行研究と比べてどこがすごい？

- Social Media の feedback 情報 (ある返答に対する返信の多さ/それ以降のツリーの長さ/Upvote の数) を用いて魅力的な対話を評価する手法の提案は初?
- 直接質に直結しない feedback を上手く使って返答の評価につなげている。

## 3. 技術や手法のキモはどこ？

- DialogRPT
    - DialoGPT ベースのモデルで、最後のトークンから linear を挟んでスコアを出力する
    - 3つの feedback それぞれについて学習したモデルと Human-like モデルを作成し、アンサンブルすることで精度向上
    - Human-like モデルはよくあるgold labelか偽データか当てるように学習したモデル
- Training
    - Social Media の feedback を直接学習するのはノイズが多く難しいので、2つの同じコンテキストに対する返答のどちらが良いかを予測するように contrastive training を行う
    - この際、ペアの作り方はかなり工夫しており、feedback の増減の他の要因がなるべく減るようにしている
    - ロスだけが contrastive で、モデルの出力自体はスコアがスカラーで得られる

## 4. どうやって有効だと検証した？

Social Media データで、いくつかの指標とともに実験。基本的に提案手法である DialogRPT が最も良いスコアとなっている。

- 元の3つの feedback をどれだけ予測できているかは、圧倒的に提案手法が相関が高い(それはそう)。
- 他のデータセットに転移してもそこそこの改善。
- 最終的に"良い返答"かどうかが評価できているかは判断しづらく、アンサンブルの重みを test set に fit しているような記述がある。

![スクリーンショット 2020-12-03 2 01 09](https://user-images.githubusercontent.com/1255485/100905483-78a62980-350b-11eb-8316-5efbbc43f9e4.png)

## 5. 議論はある？

- [IMO] 結局人の評価と近いのかどうかがちゃんと図られてない気がする。Social media の feedback は人手評価と相関はあるだろうけど高くなさそうな印象。そこはサクッと飛ばされている。
    - W_K をテストデータで決めているならちゃんとした評価になっていなさそう。
- feedback がオンラインで求められるので、対話生成モデルと組合せて品質向上が可能
    - beam search のデコーディングに組み込む
    - 強化学習の reward として使う
